invenio_github/views/badge.py

```diff
@@ -41,34 +42,6 @@
 )
 
 
-# Kept for backward compatibility
-def get_pid_of_latest_release_or_404(**kwargs):
```

Comment:
The issue here was that all of these were copy / pasted from legacy and were not properly adapted to the new architecture. E.g.:

- Badge [title](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/services/github/release.py#L173): will be `DOI` if datacite is enabled (like on Zenodo)
- Badge [value](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/services/github/release.py#L179): will take the DOI identifier from record pids if Datacite is enabled
- Record [url](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/services/github/release.py#L163): will be either the record DOI URL  if the record has one, otherwise it's HTML url

Reply:
If we use these, which are already in place, we can remove this legacy logic based on `pid` and some hardcoded stuff (e.g. doi URL)